languages/javalang/builder-renderer/commercetools-java-client/src/main/java/client/ApiHttpClient.java

```diff
@@ -1,9 +1,45 @@
 package client;
 
-import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionException;
 
-public interface ApiHttpClient {
+public class ApiHttpClient {
```

Comment:
It's now a class because the http implementation detials are handled by the middlewares?

Reply:
exactly, the real business now is happening in the middlewares, the soul purpose of this class is to do the wiring.